Sourcery/Parsing/Parser.swift

```diff
@@ -364,16 +348,52 @@ final class Parser {
         }
 
         for (_, type) in unique {
+            //find actual variables types
             for variable in type.variables {
-                if let actualTypeName = typeName(for: variable.unwrappedTypeName, containingType: type) {
+                if let actualTypeName = typeName(for: variable.unwrappedTypeName, containingType: type, typealiases: typealiases) {
                     variable.type = unique[actualTypeName]
                 } else {
                     variable.type = unique[variable.unwrappedTypeName]
                 }
             }
-        }
+            // find actual methods parameters types and their argument labels
+            for method in type.methods {
+                let argumentLabels: [String]
+                if let labels = method.fullName.range(of: "(")
+                    .map({ method.fullName.substring(from: $0.upperBound) })?
+                    .trimmingCharacters(in: CharacterSet(charactersIn: ")"))
+                    .components(separatedBy: ":")
+                    .dropLast() {
+                    argumentLabels = Array(labels)
+                } else {
+                    argumentLabels = []
+                }
 
-        for (_, type) in unique {
+                for (index, parameter) in method.parameters.enumerated() {
+                    if argumentLabels.count > index {
+                        parameter.argumentLabel = argumentLabels[index]
+                    }
+                    
+                    if let actualTypeName = typeName(for: parameter.unwrappedTypeName, containingType: type, typealiases: typealiases) {
+                        parameter.type = unique[actualTypeName]
+                    } else {
+                        parameter.type = unique[parameter.unwrappedTypeName]
+                    }
+                }
+                
+                if method.returnTypeName != "Void" && method.returnType == nil {
```

Comment:
I think it's easier to do it here, so that we already have a final type in the `returnType`, instead of for example extension if this method is contained in extension and then needing to post process it again.